lib/elixir/test/elixir/version_test.exs

```diff
@@ -317,8 +317,10 @@ defmodule VersionTest do
     {:ok, req} = Version.parse_requirement("1.2.3")
     assert req == Version.compile_requirement(req)
 
+    string = Process.get(:unused, "~> 1.2.3")
+
     assert_raise(FunctionClauseError, fn ->
-      Version.compile_requirement("~> 1.2.3")
+      Version.compile_requirement(string)
     end)
```

Comment:
In my opinion we should remove this one but it is also fine to leave it.

Reply:
Yeah I hesitated, I like the idea of ensuring the runtime behavior in such a case but happy to just remove if you prefer 👍